From f1b604e1aedbb619735128a798602ea727b94cfe Mon Sep 17 00:00:00 2001
From: Christopher Champagne <cjchampagne@habana.ai>
Date: Mon, 09 Jun 2025 15:32:48 +0300
Subject: [PATCH] [SW-222861] g3: pldm: mask subsensors by platform

Some state sensors have sub-sensors that should be supported on
HL325 and not HL338. Mask select sub-sensors of the following:
	Gaudi3 power and performance state
	Gaudi3 OAM last power down cause 1
	Gaudi3 OAM last power down cause 4
	Gaudi3 OAM last power down cause 5
	Gaudi3 OAM last power down cause 7

Change-Id: Ibd915a9e439a94f00d86305d4b0c4e2ab3f9a5f1
Signed-off-by: Christopher Champagne <cjchampagne@habana.ai>
---

diff --git a/arch/gaudi3/ras/pldm/pldm_pdr.c b/arch/gaudi3/ras/pldm/pldm_pdr.c
index 4abdb57..61d348d 100644
--- a/arch/gaudi3/ras/pldm/pldm_pdr.c
+++ b/arch/gaudi3/ras/pldm/pldm_pdr.c
@@ -981,6 +981,17 @@
 					BYTE0(OAM_LAST_POWER_DOWN_POSSIBLE_STATES_MASK),
 			},
 		},
+		.composite_unsupported_platform_mask = {
+			/* composite sensor masks for 8 sensors */
+			PLDM_PLATFORM_MASK_HL338,
+			PLDM_PLATFORM_MASK_NONE,
+			PLDM_PLATFORM_MASK_HL338,
+			PLDM_PLATFORM_MASK_NONE,
+			PLDM_PLATFORM_MASK_HL338,
+			PLDM_PLATFORM_MASK_NONE,
+			PLDM_PLATFORM_MASK_HL338,
+			PLDM_PLATFORM_MASK_NONE,
+		},
 	}},
 	[PLDM_RIDX_SENSOR_OAM_LAST_POWER_DOWN_CAUSE_2] = { .state_sensor = {
 		.header = {
@@ -1161,6 +1172,14 @@
 					BYTE0(OAM_LAST_POWER_DOWN_POSSIBLE_STATES_MASK),
 			},
 		},
+		.composite_unsupported_platform_mask = {
+			/* composite sensor masks for 5 sensors */
+			PLDM_PLATFORM_MASK_HL338,
+			PLDM_PLATFORM_MASK_NONE,
+			PLDM_PLATFORM_MASK_NONE,
+			PLDM_PLATFORM_MASK_NONE,
+			PLDM_PLATFORM_MASK_NONE,
+		},
 	}},
 	[PLDM_RIDX_SENSOR_OAM_LAST_POWER_DOWN_CAUSE_5] = { .state_sensor = {
 		.header = {
@@ -1220,6 +1239,16 @@
 				.possible_states[0] =
 					BYTE0(OAM_LAST_POWER_DOWN_POSSIBLE_STATES_MASK),
 			},
+		},
+		.composite_unsupported_platform_mask = {
+			/* composite sensor masks for 7 sensors */
+			PLDM_PLATFORM_MASK_NONE,
+			PLDM_PLATFORM_MASK_NONE,
+			PLDM_PLATFORM_MASK_NONE,
+			PLDM_PLATFORM_MASK_NONE,
+			PLDM_PLATFORM_MASK_HL338,
+			PLDM_PLATFORM_MASK_HL338,
+			PLDM_PLATFORM_MASK_HL338,
 		},
 	}},
 	[PLDM_RIDX_SENSOR_OAM_LAST_POWER_DOWN_CAUSE_6] = { .state_sensor = {
@@ -1340,6 +1369,15 @@
 				.possible_states[0] =
 					BYTE0(OAM_LAST_POWER_DOWN_POSSIBLE_STATES_MASK),
 			},
+		},
+		.composite_unsupported_platform_mask = {
+			/* composite sensor masks for 6 sensors */
+			PLDM_PLATFORM_MASK_NONE,
+			PLDM_PLATFORM_MASK_NONE,
+			PLDM_PLATFORM_MASK_NONE,
+			PLDM_PLATFORM_MASK_HL338,
+			PLDM_PLATFORM_MASK_HL338,
+			PLDM_PLATFORM_MASK_HL338,
 		},
 	}},
 	[PLDM_RIDX_SENSOR_OAM_LAST_POWER_DOWN_CAUSE_8] = { .state_sensor = {
@@ -2225,6 +2263,16 @@
 				.possible_states[0] = HBM_THROTTLING_POSSIBLE_STATES_MASK_BYTE0,
 			},
 		},
+		.composite_unsupported_platform_mask = {
+			/* composite sensor masks for 7 sensors */
+			PLDM_PLATFORM_MASK_NONE,
+			PLDM_PLATFORM_MASK_NONE,
+			PLDM_PLATFORM_MASK_NONE,
+			PLDM_PLATFORM_MASK_HL338,
+			PLDM_PLATFORM_MASK_NONE,
+			PLDM_PLATFORM_MASK_NONE,
+			PLDM_PLATFORM_MASK_NONE,
+		},
 	}},
 	[PLDM_RIDX_SENSOR_SECURITY_KEY_STATE] = { .state_sensor = {
 		.header = {
